### PR TITLE
Integration tests for text_input, text, boolean

### PR DESCRIPTION
**Pull request by dqnykamp**

### CODE REVIEW

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/text_input/test/basic_actions.rs [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    );
+    assert_eq!(get_value(text_input_idx, &mut core), "hello there");
+}
+
+/// When have two grand children, `UpdateImmediateValue` still works,
+/// but `UpdateValue` reverts both to their original value
+#[test]
+fn value_and_immediate_value_with_two_grandchildren_respond_to_actions() {
+    let dast_root =
+        dast_root_no_position(r#"<textInput ><text><text>hello</text> there</text></textInput>"#);
```
Comment (siefkenj) [inline packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/text_input/test/basic_actions.rs:179]: `<textInput >`? Random space?

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/test.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#[test]
+fn boolean_sv_is_alias_of_value() {
+    let dast_root = dast_root_no_position(r#"<boolean>true</boolean>"#);
+    let mut core = DoenetMLCore::new(dast_root, "", "", None);
+
+    // the boolean will be index 1, as the document tag will be index 0.
+    let boolean_idx = 1;
+
+    assert_eq!(get_value_prop(boolean_idx, &mut core), true);
+    assert_eq!(get_boolean_prop(boolean_idx, &mut core), true);
```
Comment (siefkenj) [inline packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/test.rs:18]: The name of this test seems to suggest that you're asserting their equal to each other, not to true necessarily.

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#[derive(Debug, ComponentState)]
+pub struct BooleanState {
+    /// The value of the `<boolean>` component.
+    ///
+    /// It is marked `is_public` so that it can be referenced in DoenetML via `.value`.
+    ///
+    /// It is marked `for_renderer` to send this value to the renderer of the `<boolean>` component.
+    ///
+    /// It is marked with a component profile state variable, indicating that the `<boolean>` component
+    /// can represent a boolean value by returning the value of this state variable.
```
Comment (siefkenj) [inline packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs:46]: I don't understand this sentence.

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    #[is_public]
+    #[for_renderer]
+    #[component_profile_state_variable]
+    value: StateVar<bool>,
+
+    /// An alias to the `value` state variable.
+    ///
+    /// It is marked `is_public` so that it can be referenced in DoenetML via `.boolean`.
```
Comment (siefkenj) [inline packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs:54]: _Somewhere_, we want clear documentation of what `#[is_public]` means, but we don't need to repeat it everywhere.

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs (hunk 3 above)
Comment (dqnykamp) [reply packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs:46]: I tried phrasing it in terms of a parent querying children to see if that is clearer.

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs (hunk 4 above)
Comment (dqnykamp) [reply packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/component.rs:54]: Yeah, just not sure where the *somewhere* is. We don't yet have a place where we are documenting how state works in general. Maybe in the definition of the `ComponentState` trait?

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/test.rs (hunk 2 above)
Comment (dqnykamp) [reply packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean/test.rs:18]: fair enough

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/text_input/test/basic_actions.rs (hunk 1 above)
Comment (dqnykamp) [reply packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/text_input/test/basic_actions.rs:179]: I don't know, but it was in there consistently, presumably due to copy-paste.

File: packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/boolean_to_string_state_var.test.rs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        queries,
+        vec![DataQuery::StateVar {
+            component_idx: None,
+            state_var_idx: 3
+        },]
+    );
+}
+
+#[test]
+fn boolean_to_string_state_var_calculated_() {
```
[the remark this answers is not in the document]
Comment (dqnykamp) [reply packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/boolean_to_string_state_var.test.rs:20]: I'm pretty sure `boolean_to_string_state_var_calculated` isn't a reserved word so it's not necessary like for `macro_`.  :)

**Pull request merged**